### ISSUE TITLE
Allowlist is not working

### ISSUE THREAD

**Issue by suk69** [SelfControlApp/selfcontrol#722]
# Summary
All websites work even though they're not on my allowlist.

In a sentence or two, what happened?
I enabled the lock and just tried browsing and I was able to use everything.

# Background Info
-

SelfControl version? 4.0 (Oct 3 update)
macOS version? 11n
<img width="1440" alt="Screenshot 2021-10-04 at 7 50 31 AM" src="https://user-images.githubusercontent.com/91859525/135784222-9e3bed88-bb5d-48a8-857e-7d33b06c3977.png">
Do you use a VPN/proxy, and if so which one? none

Has this _ever_ worked for you on a prior version of SelfControl? If so, when? Everything was fine before this update
Any other special system setup we should know about?

# How Do We Reproduce This Issue?

Describe detailed steps that reliably make this issue show up. For example, "start a block containing 5 websites, click button X, then button Y"

# What _Should_ Have Happened?

What would this have looked like if it had worked? What did you expect to happen?

# What Actually Happened Instead?

What did you actually see? e.g. what error messages popped up, what window is showing what text, etc.

**Comment by cstigler** [comment]
Hi @suk69, thanks for the report. I looked at it and since the underlying blocking mechanism didn't change, there's really two options for what's happening here. I'd love your help figuring out which is the cause of your issues.

First, can you clarify what you mean by "able to use everything"? In your screenshot, you show YouTube working, but are _all_ other sites you've tested loading? For example, if you start a block, then try to open neverssl.com or github.com or techcrunch.com, what happens?

The failure modes I can think of:

1) One of the changes in how we clear browser/OS network caches has made it so that your cache isn't being cleared properly, but the block is otherwise blocking the right sites. To test this, start a SelfControl allowlist block, then quit your browser, open it up, clear your browser cache, and try accessing some sites. If it works at that point, it means it was likely a caching issues

2) It's an issue with our allowlist scraper for linked sites, which has gotten "better" since last release and could be scraping links to YouTube etc from your allowlisted sites (they're very common as social media share links, and embedded videos). To test this, go to SelfControl --> Preferences, click over to the Advanced setting, and uncheck the "Allowlist blocks should include linked sites" box. Now start a new allowlist block, and see whether you have the same issue.

**Comment by suk69** [comment]
<img width="1440" alt="Screenshot 2021-10-05 at 11 20 42 AM" src="https://user-images.githubusercontent.com/91859525/135967699-3d92d731-1908-4ea5-8030-edbe92b09af2.png"
<img width="1440" alt="Screenshot 2021-10-05 at 11 20 56 AM" src="https://user-images.githubusercontent.com/91859525/135967719-b7e30a16-0bad-43e8-9ef6-11fb5d7ecb2f.png">
>

Tried both the methods, and everything works, as in every site is still working. I'll be uninstalling the software once the timer ends and then reinstall it, because well, I'm no techie lol. Maybe it's a bug, which is only happening to me. If the lock doesn't work again, I'll be going to the sweet ol' 3.9 version (if any, the one before 4.0).
Thank you for your reply, I'll be looking forward to another one.

**Comment by suk69** [comment]
I forgot to mention this, thanks for your awesome tool. It's helped me out a lot in the past. 
I use it so much, I'm using it while writing this. Just kidding. Thanks for the tool guys.

**Comment by UncleSeb** [comment]
I am having the exact same issue. Everything worked fine until 4.0, now the allowlist does not work.

**Comment by mcvmcvmcvmcv** [comment]
Same for me too--with sites in the allowlist, 4.0.1 does not block anything. When I delete everything from the allowlist, i.e. block the entire internet, then the block works. Happy to provide more info if needed!

**Comment by turnwrite** [comment]
I am having this same issue; with sites on my allowlist, SelfControl will not block anything. Is there a way to fix this yet?

**Comment by illuninati** [comment]
Having the same problem, trying allow list for the first time and can't seem to get it to work. 
SelfControl 4.0.2, MacOS 11.6.1, Chrome 96.0.4664.110

**Comment by michaelgrabowy** [comment]
Same issue here. This is on OS 10.14.6 and SC version 4.0.2. As long as there is anything on the allowlist, nothing is blocked. Restarting browsers does not help.

**Comment by Marshpillow** [comment]
Same issue here, allowlist is not working.

**Comment by LukeTully** [comment]
I'm finding that, as mentioned in a previous issue, blocklist does not seem to be working in Firefox with DNS-over-HTTPS enabled. Turning that off works. I guess that's fundamentally contrary to how SelfControl probably works hey?

**Comment by audarya108** [comment]
Has anyone resolved this problem of the "Allow List" not working with Mac 12.6 (that's my case)?

**Comment by mcvmcvmcvmcv** [comment]
No, I have stopped using the app as a result. I just turn off my wifi and live with the results lol

**Comment by audarya108** [comment]
I wonder why the developers can't adjust/tweak it for the benefits of all for whom they/he/she made the app in the 1st place!

**Comment by slambert** [comment]
> I wonder why the developers can't adjust/tweak it for the benefits of all for whom they/he/she made the app in the 1st place!

Just a reminder that "the developers" are all volunteers. I think you are mistaking this project for a product. A product is usually accompanied by customer service. Behaving as if SelfControl is a product means expecting “customer service” to fix it, to bring satisfaction, and to do both quickly. However, free/open source software does not work according to the customer service paradigm. People can help by submitting good bug reports, submitting code, hiring someone if they do not have the skills, or otherwise being patient and kind.

**Comment by MarfisaLib** [comment]
Same issue here. Have we figured out what's causing it? 

I disabled my icloud private relay and Netgear shield. Restarted the browser and restarted the computer. Deleted and redownloaded the app. 

The blocklist function works, and if there's nothing in the allowlist then all websites are blocked. But if I have anything in the allowlist, all websites are allowed.

**Comment by Fivapr** [comment]
use 3.0.3 instead
